pallets/gear/src/lib.rs

```diff
@@ -1690,6 +1716,8 @@ pub mod pallet {
                     Error::<T>::InactiveProgram
                 );
 
+                Self::check_gas_limit(gas_limit)?;
```

Comment:
why moved below?

Reply:
We discussed it in DM before. If a message is not for a program then it won't be enqueued hence no point to check gas limit (checking value is being deprecated, as well).